src/DssSpell.t.sol

```diff
@@ -1111,4 +1120,51 @@ contract DssSpellTest is DSTest, DSMath {
         assertEq(MedianAbstract(UNIUSD_MED).bud(SET_UNI), 1);
     }
 
+    event Log(string message, address deployer, bytes32 contractName);
+
+    function hasWards(address addr) internal returns (bool) {
+        (bool ok, bytes memory data) = addr
+            .call(abi.encodeWithSignature("wards(address)", 0x0));
+        return ok && data.length == 32;
+    }
+
+    address[] evilAddresses = [
+        0xdDb108893104dE4E1C6d0E47c42237dB4E617ACc,
+        0xDa0FaB05039809e63C5D068c897c3e602fA97457,
+        0xda0fab060e6cc7b1C0AA105d29Bd50D71f036711,
+        0xDA0FaB0700A4389F6E6679aBAb1692B4601ce9bf
+    ];
+
+    function hasBadAuth(address addr, bytes32 contractName) internal {
+        for (uint i = 0; i < evilAddresses.length; i ++) {
+            if (AuthLike(addr).wards(evilAddresses[i]) > 0) {
+                emit Log("Bad auth", evilAddresses[i], contractName);
+                fail();
+            }
+        }
+    }
+
+    function test_wards() public {
+        vote();
+        spell.schedule();
+        castPreviousSpell();
+        hevm.warp(spell.nextCastTime());
+        spell.cast();
+        assertTrue(spell.done());
+        ChainLogLike chainLog = ChainLogLike(addr.addr("CHANGELOG"));
+        bytes32[] memory contractNames = chainLog.list();
+        for(uint i = 0; i < contractNames.length; i++) {
+            try chainLog.getAddress(contractNames[i]) returns (address addr) {
```

Comment:
why do we need the try catch? Isn't supposed if the list brings a name that will be a valid key?

Reply:
I thought so too, but I was having issues with `FAUCET`. I guess it makes sense for it to be the zero address on mainnet.

Reply:
Yeah, we might want to remove "FAUCET" from the mainnet changelog.

Reply:
Discussed offline. We're planning to remove `FAUCET` from the changelog.